### PR TITLE
new implementation of mumps wrapper

### PR DESCRIPTION
**Pull request by vlukes**
Testing on Windows machines needed!

### CODE REVIEW

**Comment by rc** [comment]
Thanks! For windows, wait for the AppVeyor build :)

File: sfepy/solvers/ls.py [hunk 1]
```diff
@@ -746,51 +746,110 @@ class MUMPSSolver(LinearSolver):
...
-        except ImportError:
-            self.mumps = None
-            msg = 'cannot import MUMPS!'
-            raise ImportError(msg)
+        import sfepy.solvers.ls_mumps as mumps
 
-        LinearSolver.__init__(self, conf, **kwargs)
-        self.mumps = DMumpsContext()
-        self.mumps_presolved = False
+        self.mumpy_ls = None
```
Comment (rc) [inline sfepy/solvers/ls.py]: `self.mumpy_ls -> self.mumps_ls`

**Comment by rc** [comment]
There are many test failures due to
```
  File "/home/travis/build/sfepy/sfepy/sfepy/solvers/ls_mumps.py", line 2, in <module>
    from mpi4py import MPI
ImportError: No module named mpi4py
```
Could you hide the MPI import? E.g. using
```
if __name__ == "__main__":
    main()
```

On my computer, with MPI installed, I am getting:
```
ImportError: No library named dmumps
```
failure in `load_library()` - the problem is, that I do not have the paths to system libraries in `LD_LIBRARY_PATH` variable. 

What works for me is (on linux):
```
       from ctypes.util import find_library
        lib_fname = find_library(libname)
        lib = ctypes.cdll.LoadLibrary(lib_fname)
```
Note that the Windows branch has a bug (remove `''` in `find_library('libname')`).

So, after all, maybe the windows and linux branches could be the same?

**Comment by vlukes** [comment]
> There are many test failures due to
> 
>   File "/home/travis/build/sfepy/sfepy/sfepy/solvers/ls_mumps.py", line 2, in <module>
>     from mpi4py import MPI
> ImportError: No module named mpi4py
> 
> Could you hide the MPI import? E.g. using
> 
> if __name__ == "__main__":
>     main()

I used the following to avoid test failures:
```
try:
    from mpi4py import MPI
    use_mpi = True
except ImportError:
    use_mpi = False
```
> On my computer, with MPI installed, I am getting:
> 
> ImportError: No library named dmumps

You need the library, so install `libmumps-dev` on Ubuntu.
 
> failure in load_library() - the problem is, that I do not have the paths to system libraries in LD_LIBRARY_PATH variable.
> 
> What works for me is (on linux):
> 
>        from ctypes.util import find_library
>         lib_fname = find_library(libname)
>         lib = ctypes.cdll.LoadLibrary(lib_fname)
>
> Note that the Windows branch has a bug (remove '' in find_library('libname')).

Thanks, debugged.
 
> So, after all, maybe the windows and linux branches could be the same?

There is the following issue (see, [1]), `find_library` returns:
- only the filename of the library file on Linux
- the full pathname on Windows

That is the reason for two branches in the code.

[1] https://docs.python.org/2/library/ctypes.html

**Comment by rc** [comment]
On 05/04/2018 02:54 PM, Vladimir Lukes wrote:
>> There are many test failures due to
>>
>>    File "/home/travis/build/sfepy/sfepy/sfepy/solvers/ls_mumps.py", line 2, in <module>
>>      from mpi4py import MPI
>> ImportError: No module named mpi4py
>>
>> Could you hide the MPI import? E.g. using
>>
>> if __name__ == "__main__":
>>      main()
> 
> I used the following to avoid test failures:
> ```
> try:
>      from mpi4py import MPI
>      use_mpi = True
> except ImportError:
>      use_mpi = False
> ```

Thanks, Travis tests now pass.

>> On my computer, with MPI installed, I am getting:
>>
>> ImportError: No library named dmumps
> 
> You need the library, so install `libmumps-dev` on Ubuntu.

I have it installed. The problem is below:

>> failure in load_library() - the problem is, that I do not have the paths to system libraries in LD_LIBRARY_PATH variable.
>>
>> What works for me is (on linux):
>>
>>         from ctypes.util import find_library
>>          lib_fname = find_library(libname)
>>          lib = ctypes.cdll.LoadLibrary(lib_fname)
>>
>> Note that the Windows branch has a bug (remove '' in find_library('libname')).
> 
> Thanks, debugged.
>   
>> So, after all, maybe the windows and linux branches could be the same?
> 
> There is the following issue (see, [1]), `find_library` returns:
> - only the filename of the library file on Linux
> - the full pathname on Windows
> 
> That is the reason for two branches in the code.
> 
> [1] https://docs.python.org/2/library/ctypes.html
> 

On my linux box, the windows branch works, but the linux one does not. Does the 
windows branch work on your linux box?

**Comment by rc** [comment]
@vlukes, any update?

**Comment by vlukes** [comment]
I have changed the way of the loading library. Could you try it on your machine?

**Comment by vlukes** [comment]
Yes, of course. Fixed now.

**Comment by rc** [comment]
It works both on my laptop and desktop. I will merge it after travis tests finish. Thanks!

**Pull request merged**